src/components/clusterDetail/ConsoleModal.tsx

```diff
@@ -77,34 +81,102 @@ const getHostIPs = (cluster: Cluster): { [key: string]: string } => {
   return hostIPAddresses;
 };
 
+const getVirtualIPs = ({
```

Comment:
a `Cluster` is passed here, we can destructuralize later and so avoid typescript's `Partial`

Reply:
sorry I don't understand what you mean

Reply:
when calling that function, a `cluster` object is passed

Reply:
```suggestion
const getVirtualIPs = (cluster: Cluster) => {
  const { apiVip, ingressVip, ./* TBD*/... } = cluster;
  
  ....
}
```